src/components/Authentication/Authentication.vue

```diff
@@ -0,0 +1,67 @@
+<template lang="pug">
+.girder-authentication-component
+  v-card
+    v-tabs(v-model="activeTab", color="primary", slider-color="yellow", dark)
+      v-tab(key="login") Log In
+      v-tab(key="registration", v-if="register") Register
+      v-tab-item(key="login-box")
+        girder-login(:oauth-providers="oauthProviders",
+            :forgot-password-url="forgotPasswordUrl",
+            :forgot-password-route="forgotPasswordRoute",
+            @forgotpassword="$emit('forgotpassword')")
+      v-tab-item(key="registration-box", v-if="register")
+        girder-registration(:oauth-providers="oauthProviders")
+</template>
+
+<script>
+import GirderLogin from './Login.vue';
+import GirderRegistration from './Register.vue';
+
+export default {
+  inject: ['girderRest'],
+  props: {
+    /* Enable registration from this component? */
+    register: {
+      type: Boolean,
+      default: false,
+    },
+    /* Enable OAuth login from this component? */
+    oauth: {
+      type: Boolean,
+      default: false,
+    },
+    /* A full URL to be used as an anchor href to an external page.
+     * Overrides forgotPasswordRoute.
+     */
+    forgotPasswordUrl: {
+      type: String,
+    },
+    /* A vue-router route path.  Must use object form. */
+    forgotPasswordRoute: {
+      type: Object,
```

Comment:
`type: [Object, String]`, since a route can be expressed as either type and the underlying components support either.